### PR TITLE
WIP - Mssql fixes

### PR DESCRIPTION
**Pull request by cedriclombardot**

### CODE REVIEW

File: src/Propel/Runtime/Connection/ConnectionFactory.php [hunk 1]
```diff
@@ -55,6 +55,11 @@ public static function create(array $configuration = [], AdapterInterface $adapt
                     }
                     $value = constant($value);
                 }
+                
+                if ($option == "ATTR_EMULATE_PREPARES" || $option == "ATTR_TIMEOUT") {
```
Comment (cedriclombardot) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: How set it per db type ?

Comment (marcj) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: Why would you exclude those?

Comment (cedriclombardot) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: Beacause when we use it in mssql it fail (Eg : http://stackoverflow.com/questions/16366709/php-pdo-sqlsrv-cant-set-logintimeout)

Comment (marcj) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: well, then the developer can simply not use these options?

Comment (cedriclombardot) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: They was added automatically in propel https://github.com/propelorm/Propel2/blob/97dcdae4aff3ae6ae5f5bcf356a72babaa9f293d/src/Propel/Common/Config/PropelConfiguration.php#L132

Comment (marcj) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: Oh, then we should change that :)

**Comment by cedriclombardot** [comment]
Yes, but i dont know how do this properly without breaking mysql feature

File: src/Propel/Runtime/Connection/ConnectionFactory.php (hunk 1 above)
Comment (marcj) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: ok, then maybe the check is better placed in Connection::setAttribute() which has the DSN and knows whether its MSSQL or not

Comment (marcj) [inline src/Propel/Runtime/Connection/ConnectionFactory.php]: This needs to be moved first until I can merge it :)

File: src/Propel/Runtime/Adapter/Pdo/MssqlAdapter.php [hunk 2]
```diff
@@ -300,4 +300,25 @@ public function cleanupSQL(&$sql, array &$params, Criteria $values, DatabaseMap
...
     }
+    
+    /**
+     * Returns timestamp formatter string for use in date() function.
+     *
+     * @return string
+     */
+    public function getTimestampFormatter()
+    {
+        return 'Y-m-d H:i:s:000';
```
Comment (Incognito) [inline src/Propel/Runtime/Adapter/Pdo/MssqlAdapter.php:311]: Can MSSSQL support `Y-m-d H:i:s.u` or is `:000` required?

Comment (DarkAxi0m) [reply src/Propel/Runtime/Adapter/Pdo/MssqlAdapter.php:311]: Just finished some testing and was going to create a PR, when I noticed this.

DateTime   can use `Y-m-d H:i:s.u`;
But DateTime2 needs to use `Y-m-d H:i:s`;

_(not sure how to proceeded from here)_

**Comment by dereuromark** [comment]
Replaced by https://github.com/propelorm/Propel2/pull/1881

**Pull request closed without merging**